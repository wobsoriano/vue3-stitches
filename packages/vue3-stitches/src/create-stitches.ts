import type { ConfigType, DefaultThemeMap } from '@stitches/core/types/config'
import type * as CSSUtil from '@stitches/core/types/css-util'
import type * as Util from '@stitches/core/types/util'
import type { RemoveIndex } from '@stitches/core/types/stitches'
import type Stitches from '@stitches/core/types/stitches'
import { createStitches as createStitchesCore } from '@stitches/core'
import { defineComponent, h } from 'vue'
import type * as StyledComponent from './styled-component'

export interface EmptyObject {}

export interface VueStitches <Prefix extends string = '', Media = EmptyObject, Theme = EmptyObject, ThemeMap = EmptyObject, Utils = EmptyObject> extends
  Stitches<Prefix, Media, Theme, ThemeMap, Utils> {
  styled: {
    <
      Type extends keyof StyledComponent.IntrinsicElementAttributes,
      Composers extends (
        | string
        | Util.Function
        | Record<string, unknown>
      )[],
      CSS = CSSUtil.CSS<Media, Theme, ThemeMap, Utils>,
    >(
      type: Type,
      ...composers: {
        [K in keyof Composers]: (
          string extends Composers[K]
            ? Composers[K]
            : Composers[K] extends string| Util.Function
              ? Composers[K]
              : RemoveIndex<CSS> & {
                variants?: {
                  [Name in string]: {
                    [Pair in number | string]: CSS
                  }
                }
                compoundVariants?: (
                  & (
                    'variants' extends keyof Composers[K]
                      ? {
                          [Name in keyof Composers[K]['variants']]?: Util.Widen<keyof Composers[K]['variants'][Name]> | Util.String
                        }
                      : Util.WideObject
                  )
                  & {
                    css: CSS
                  }
                )[]
                defaultVariants?: (
                  'variants' extends keyof Composers[K]
                    ? {
                        [Name in keyof Composers[K]['variants']]?: Util.Widen<keyof Composers[K]['variants'][Name]> | Util.String
                      }
                    : Util.WideObject
                )
              } & CSS & {
                [K2 in keyof Composers[K]]: K2 extends 'compoundVariants' | 'defaultVariants' | 'variants'
                  ? unknown
                  : K2 extends keyof CSS
                    ? CSS[K2]
                    : unknown
              }
        )
      }
    ): StyledComponent.StyledComponent<
      Type,
      StyledComponent.StyledComponentProps<Composers>,
      Media,
      CSSUtil.CSS<Media, Theme, ThemeMap, Utils>
    >
  }
}

export const createStitches = <Prefix extends string = string,
  Media extends EmptyObject = EmptyObject,
  Theme extends EmptyObject = EmptyObject,
  ThemeMap extends EmptyObject = DefaultThemeMap,
  Utils extends EmptyObject = EmptyObject>(config?: {
    media?: ConfigType.Media<Media>
    prefix?: ConfigType.Prefix<Prefix>
    theme?: ConfigType.Theme<Theme>
    themeMap?: ConfigType.ThemeMap<ThemeMap>
    utils?: ConfigType.Utils<Utils>
  }): VueStitches<Prefix, Media, Theme, ThemeMap, Utils> => {
  const stitchesCore = createStitchesCore(config)
  const { css } = stitchesCore

  const styled = (tag: any, composers: any) => {
    const result = css(composers)
    return defineComponent({
      setup(_props, { slots, attrs }) {
        return () => {
          const { className } = result(attrs)
          const classes = [className]

          return h(tag, {
            class: classes,
          }, slots)
        }
      },
    })
  }

  return {
    ...stitchesCore,
    // @ts-expect-error: Fix later
    styled,
  }
}
