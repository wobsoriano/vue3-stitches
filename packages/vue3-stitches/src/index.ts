import type * as CSSUtil from '@stitches/core/types/css-util'
import type * as Config from '@stitches/core/types/config'
import type { VueStitches } from './features/styled'
import { getCachedConfig } from './utility/getCachedConfig'
import type * as StyledComponent from './features/styled-component'

export type { $$PropertyValue, $$ScaleValue, $$ThemeValue } from '@stitches/core'

export type CreateStitches = Config.CreateStitches
export type CSSProperties = CSSUtil.CSSProperties
export type DefaultThemeMap = Config.DefaultThemeMap
export type FontFace = CSSUtil.Native.AtRule.FontFace

/** Returns a Style interface from a configuration, leveraging the given media and style map. */
export type CSS<
  Config extends {
    media?: {}
    theme?: {}
    themeMap?: {}
    utils?: {}
  } = {
    media: {}
    theme: {}
    themeMap: {}
    utils: {}
  },
  > = CSSUtil.CSS<
  Config['media'],
  Config['theme'],
  Config['themeMap'],
  Config['utils']
>

/** Returns the properties, attributes, and children expected by a component. */
export type ComponentProps<Component> = Component extends ((...args: any[]) => any) ? Parameters<Component>[0] : never

/** Returns a type that expects a value to be a kind of CSS property value. */
export type PropertyValue<Property extends keyof CSSUtil.CSSProperties, Config = null> = (
  Config extends null
    ? { readonly [K in CSSUtil.$$PropertyValue]: Property }
    : Config extends Record<string, any>
      ? CSSUtil.CSS<
      Config['media'],
      Config['theme'],
      Config['themeMap'],
      Config['utils']
    >[Property]
      : never
)

/** Returns a type that expects a value to be a kind of theme scale value. */
export type ScaleValue<Scale, Config = null> = (
  Config extends null
    ? { readonly [K in CSSUtil.$$ScaleValue]: Scale }
    : Config extends Record<string, any>
      ? Scale extends keyof Config['theme']
        ? `$${string & keyof Config['theme'][Scale]}`
        : never
      : never
)

/** Returns a type that suggests variants from a component as possible prop values. */
export type VariantProps<Component extends Record<symbol | string, any>> = StyledComponent.TransformProps<Component[StyledComponent.$$StyledComponentProps], Component[StyledComponent.$$StyledComponentMedia]>

/** Map of CSS properties to token scales. */
export { defaultThemeMap } from '@stitches/core'

/** Returns a library used to create styles. */
export { createStitches } from './features/styled'

/** Returns an object representing a theme. */
export const createTheme: VueStitches['createTheme'] = (...args: any[]) => getCachedConfig().createTheme(...args)

/** Returns a function that applies globalCss styles. */
export const globalCss: VueStitches['globalCss'] = (...args: any[]) => getCachedConfig().globalCss(...args)

/** Returns an object that applies `@keyframes` styles. */
export const keyframes: VueStitches['keyframes'] = (...args: any[]) => getCachedConfig().keyframes(...args)

/** Returns a function that applies styles and variants for a specific class. */
export const css: VueStitches['css'] = (...args: any[]) => getCachedConfig().css(...args)

/** Returns a function that applies styles and variants for a specific class. */
export const styled: VueStitches['styled'] = (...args: any[]) => getCachedConfig().styled(...args)

export {
  default as TestComponent,
} from './test'
