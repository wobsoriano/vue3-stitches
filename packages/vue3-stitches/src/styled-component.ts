import type * as Util from '@stitches/core/types/util'
import type * as Vue from 'vue'

export type IntrinsicElementsKeys = keyof IntrinsicElementAttributes

type __VLS_NonUndefinedable<T> = T extends undefined ? never : T
type __VLS_TypePropsToRuntimeProps<T> = {
  [K in keyof T]-?: {} extends Pick<T, K> ? {
    type: Vue.PropType<__VLS_NonUndefinedable<T[K]>>
  } : {
    type: Vue.PropType<T[K]>
    required: true
  };
}

export type StyledComponent<
  Type extends IntrinsicElementsKeys | Vue.Component | Util.Function = 'span',
  Props = {},
  Media = {},
  CSS = {},
  InnerProps = {},
> = Vue.DefineComponent<
  __VLS_TypePropsToRuntimeProps<TransformProps<Util.Assign<InnerProps, Props>, Media> & { css?: CSS }>,
  {},
  {},
  Vue.ComputedOptions,
  Vue.MethodOptions,
  Vue.ComponentOptionsMixin,
  Vue.ComponentOptionsMixin,
  Record<string, any>,
  string,
  Vue.VNodeProps & Vue.AllowedComponentProps & Vue.ComponentCustomProps,
  Vue.ExtractPropTypes<
    __VLS_TypePropsToRuntimeProps<TransformProps<Util.Assign<InnerProps, Props>, Media> & { css?: CSS }>
  > &
  (Type extends IntrinsicElementsKeys
    ? IntrinsicElementAttributes[Type]
    : {}),
  {}
> & {
  className: string
  selector: string

  [$$StyledComponentType]: Type
  [$$StyledComponentProps]: Props
  [$$StyledComponentMedia]: Media
}

/** Returns a new CSS Component. */
export interface CssComponent<
  Type = 'span',
  Props = {},
  Media = {},
  CSS = {},
> {
  (
    props?:
    & TransformProps<Props, Media>
    & {
      css?: CSS
    }
    & {
      [name in number | string]: any
    }
  ): string & {
    className: string
    selector: string
    props: {}
  }

  className: string
  selector: string

  [$$StyledComponentType]: Type
  [$$StyledComponentProps]: Props
  [$$StyledComponentMedia]: Media
}

export type TransformProps < Props, Media > = {
  [K in keyof Props]: (|
  Props[K] |
  (&
  {
    [KMedia in Util.Prefixed < '@', 'initial' | keyof Media > ]?: Props[K]
  } &
  {
    [KMedia in string]: Props[K]
  }
  )
  )
}

/** Unique symbol used to reference the type of a Styled Component. */
export declare const $$StyledComponentType: unique symbol

/** Unique symbol used to reference the type of a Styled Component. */
export type $$StyledComponentType = typeof $$StyledComponentType

/** Unique symbol used to reference the props of a Styled Component. */
export declare const $$StyledComponentProps: unique symbol

/** Unique symbol used to reference the props of a Styled Component. */
export type $$StyledComponentProps = typeof $$StyledComponentProps

/** Unique symbol used to reference the media passed into a Styled Component. */
export declare const $$StyledComponentMedia: unique symbol

/** Unique symbol used to reference the media passed into a Styled Component. */
export type $$StyledComponentMedia = typeof $$StyledComponentMedia

/** Returns a narrowed JSX element from the given tag name. */
// type IntrinsicElement<TagName> = TagName extends IntrinsicElementsKeys ? TagName : never

/** Returns the first Styled Component type from the given array of compositions. */
export type StyledComponentType < T extends any[] > = (
  T[0] extends never ?
    'span' :
    T[0] extends string ?
      T[0] :
      T[0] extends(props: any) => any ?
        T[0] :
        T[0] extends {
          [$$StyledComponentType]: unknown
        } ?
          T[0][$$StyledComponentType] :
          T extends[lead: any, ...tail: infer V] ?
            StyledComponentType < V >
            :
            never
)

/** Returns the cumulative variants from the given array of compositions. */
export type StyledComponentProps < T extends any[] > = (&
(
  $$StyledComponentProps extends keyof T[0] ?
    T[0][$$StyledComponentProps] :
    T[0] extends {
      variants: Record<string, unknown>
    } ?
        {
          [K in keyof T[0]['variants']]?: Util.Widen < keyof T[0]['variants'][K] >
        } :
        {}
) &
(
  T extends[lead: any, ...tail: infer V] ?
    StyledComponentProps < V >
    :
      {}
))

export interface IntrinsicElementAttributes {
  a: Vue.AnchorHTMLAttributes
  abbr: Vue.HTMLAttributes
  address: Vue.HTMLAttributes
  area: Vue.AreaHTMLAttributes
  article: Vue.HTMLAttributes
  aside: Vue.HTMLAttributes
  audio: Vue.AudioHTMLAttributes
  b: Vue.HTMLAttributes
  base: Vue.BaseHTMLAttributes
  bdi: Vue.HTMLAttributes
  bdo: Vue.HTMLAttributes
  blockquote: Vue.BlockquoteHTMLAttributes
  body: Vue.HTMLAttributes
  br: Vue.HTMLAttributes
  button: Vue.ButtonHTMLAttributes
  canvas: Vue.CanvasHTMLAttributes
  caption: Vue.HTMLAttributes
  cite: Vue.HTMLAttributes
  code: Vue.HTMLAttributes
  col: Vue.ColHTMLAttributes
  colgroup: Vue.ColgroupHTMLAttributes
  data: Vue.DataHTMLAttributes
  datalist: Vue.HTMLAttributes
  dd: Vue.HTMLAttributes
  del: Vue.DelHTMLAttributes
  details: Vue.DetailsHTMLAttributes
  dfn: Vue.HTMLAttributes
  dialog: Vue.DialogHTMLAttributes
  div: Vue.HTMLAttributes
  dl: Vue.HTMLAttributes
  dt: Vue.HTMLAttributes
  em: Vue.HTMLAttributes
  embed: Vue.EmbedHTMLAttributes
  fieldset: Vue.FieldsetHTMLAttributes
  figcaption: Vue.HTMLAttributes
  figure: Vue.HTMLAttributes
  footer: Vue.HTMLAttributes
  form: Vue.FormHTMLAttributes
  h1: Vue.HTMLAttributes
  h2: Vue.HTMLAttributes
  h3: Vue.HTMLAttributes
  h4: Vue.HTMLAttributes
  h5: Vue.HTMLAttributes
  h6: Vue.HTMLAttributes
  head: Vue.HTMLAttributes
  header: Vue.HTMLAttributes
  hgroup: Vue.HTMLAttributes
  hr: Vue.HTMLAttributes
  html: Vue.HtmlHTMLAttributes
  i: Vue.HTMLAttributes
  iframe: Vue.IframeHTMLAttributes
  img: Vue.ImgHTMLAttributes
  input: Vue.InputHTMLAttributes
  ins: Vue.InsHTMLAttributes
  kbd: Vue.HTMLAttributes
  keygen: Vue.KeygenHTMLAttributes
  label: Vue.LabelHTMLAttributes
  legend: Vue.HTMLAttributes
  li: Vue.LiHTMLAttributes
  link: Vue.LinkHTMLAttributes
  main: Vue.HTMLAttributes
  map: Vue.MapHTMLAttributes
  mark: Vue.HTMLAttributes
  menu: Vue.MenuHTMLAttributes
  meta: Vue.MetaHTMLAttributes
  meter: Vue.MeterHTMLAttributes
  nav: Vue.HTMLAttributes
  noindex: Vue.HTMLAttributes
  noscript: Vue.HTMLAttributes
  object: Vue.ObjectHTMLAttributes
  ol: Vue.OlHTMLAttributes
  optgroup: Vue.OptgroupHTMLAttributes
  option: Vue.OptionHTMLAttributes
  output: Vue.OutputHTMLAttributes
  p: Vue.HTMLAttributes
  param: Vue.ParamHTMLAttributes
  picture: Vue.HTMLAttributes
  pre: Vue.HTMLAttributes
  progress: Vue.ProgressHTMLAttributes
  q: Vue.QuoteHTMLAttributes
  rp: Vue.HTMLAttributes
  rt: Vue.HTMLAttributes
  ruby: Vue.HTMLAttributes
  s: Vue.HTMLAttributes
  samp: Vue.HTMLAttributes
  script: Vue.ScriptHTMLAttributes
  section: Vue.HTMLAttributes
  select: Vue.SelectHTMLAttributes
  small: Vue.HTMLAttributes
  source: Vue.SourceHTMLAttributes
  span: Vue.HTMLAttributes
  strong: Vue.HTMLAttributes
  style: Vue.StyleHTMLAttributes
  sub: Vue.HTMLAttributes
  summary: Vue.HTMLAttributes
  sup: Vue.HTMLAttributes
  table: Vue.TableHTMLAttributes
  template: Vue.HTMLAttributes
  tbody: Vue.HTMLAttributes
  td: Vue.TdHTMLAttributes
  textarea: Vue.TextareaHTMLAttributes
  tfoot: Vue.HTMLAttributes
  th: Vue.ThHTMLAttributes
  thead: Vue.HTMLAttributes
  time: Vue.TimeHTMLAttributes
  title: Vue.HTMLAttributes
  tr: Vue.HTMLAttributes
  track: Vue.TrackHTMLAttributes
  u: Vue.HTMLAttributes
  ul: Vue.HTMLAttributes
  var: Vue.HTMLAttributes
  video: Vue.VideoHTMLAttributes
  wbr: Vue.HTMLAttributes
  webview: Vue.WebViewHTMLAttributes

  // SVG
  svg: Vue.SVGAttributes

  animate: Vue.SVGAttributes
  animateMotion: Vue.SVGAttributes
  animateTransform: Vue.SVGAttributes
  circle: Vue.SVGAttributes
  clipPath: Vue.SVGAttributes
  defs: Vue.SVGAttributes
  desc: Vue.SVGAttributes
  ellipse: Vue.SVGAttributes
  feBlend: Vue.SVGAttributes
  feColorMatrix: Vue.SVGAttributes
  feComponentTransfer: Vue.SVGAttributes
  feComposite: Vue.SVGAttributes
  feConvolveMatrix: Vue.SVGAttributes
  feDiffuseLighting: Vue.SVGAttributes
  feDisplacementMap: Vue.SVGAttributes
  feDistantLight: Vue.SVGAttributes
  feDropShadow: Vue.SVGAttributes
  feFlood: Vue.SVGAttributes
  feFuncA: Vue.SVGAttributes
  feFuncB: Vue.SVGAttributes
  feFuncG: Vue.SVGAttributes
  feFuncR: Vue.SVGAttributes
  feGaussianBlur: Vue.SVGAttributes
  feImage: Vue.SVGAttributes
  feMerge: Vue.SVGAttributes
  feMergeNode: Vue.SVGAttributes
  feMorphology: Vue.SVGAttributes
  feOffset: Vue.SVGAttributes
  fePointLight: Vue.SVGAttributes
  feSpecularLighting: Vue.SVGAttributes
  feSpotLight: Vue.SVGAttributes
  feTile: Vue.SVGAttributes
  feTurbulence: Vue.SVGAttributes
  filter: Vue.SVGAttributes
  foreignObject: Vue.SVGAttributes
  g: Vue.SVGAttributes
  image: Vue.SVGAttributes
  line: Vue.SVGAttributes
  linearGradient: Vue.SVGAttributes
  marker: Vue.SVGAttributes
  mask: Vue.SVGAttributes
  metadata: Vue.SVGAttributes
  mpath: Vue.SVGAttributes
  path: Vue.SVGAttributes
  pattern: Vue.SVGAttributes
  polygon: Vue.SVGAttributes
  polyline: Vue.SVGAttributes
  radialGradient: Vue.SVGAttributes
  rect: Vue.SVGAttributes
  stop: Vue.SVGAttributes
  switch: Vue.SVGAttributes
  symbol: Vue.SVGAttributes
  text: Vue.SVGAttributes
  textPath: Vue.SVGAttributes
  tspan: Vue.SVGAttributes
  use: Vue.SVGAttributes
  view: Vue.SVGAttributes
}
