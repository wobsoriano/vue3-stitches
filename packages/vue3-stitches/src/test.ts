import { defineComponent, h } from 'vue'

export default defineComponent({
  props: {
    firstName: {
      type: String,
      required: true,
    },
    lastName: {
      type: String,
      required: false,
      default: '',
    },
  },
  setup(props) {
    return () => h('div', `${props.firstName}`)
  },
})
