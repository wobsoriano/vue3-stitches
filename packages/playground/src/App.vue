<script setup lang="ts">
import { createStitches } from 'vue3-stitches'
import type { VariantProps } from 'vue3-stitches'

const { styled, css } = createStitches()

const Btn = styled('button', {
  variants: {
    colour: {
      violet: {
        'backgroundColor': 'blueviolet',
        'color': 'red',
        '&:hover': {
          backgroundColor: 'darkviolet',
        },
      },
      gray: {
        'backgroundColor': 'gainsboro',
        '&:hover': {
          backgroundColor: 'lightgray',
        },
      },
    },
  },
})

type ASD = VariantProps<typeof Btn>

type ZXC = ASD['colour']
</script>

<template>
  <Btn>
    Hello
  </Btn>
</template>
