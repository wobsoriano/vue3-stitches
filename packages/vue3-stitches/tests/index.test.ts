import { beforeEach, describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import { defineComponent, h } from 'vue'
import { createStitches } from '../src'

let { styled } = createStitches({})

beforeEach(() => {
  const newStyled = createStitches({})
  styled = newStyled.styled
})

describe('createStitches Vue', () => {
  test('basic component', async() => {
    const Heading = styled('h1', {
      color: 'red',
    })

    // @ts-expect-error: Styled Component types incompatible
    const wrapper = mount(Heading)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('component with variant', () => {
    const Button = styled('button', {
      backgroundColor: 'gray',
      variants: {
        variant: {
          red: { backgroundColor: 'red' },
          blue: { backgroundColor: 'blue' },
        },
      },
    })

    const App = defineComponent({
      setup() {
        return () => h('div', [
          h(Button, () => 'A button!'),
          h(Button, {
            variant: 'red',
          }, () => 'A button!'),
          h(Button, {
            variant: 'blue',
          }, () => 'A button!'),
        ])
      },
    })

    const wrapper = mount(App)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('css prop', () => {
    const Button = styled('button', {
      backgroundColor: 'gray',
    })

    const App = defineComponent({
      setup() {
        return () => h(Button, {
          css: {
            color: 'red',
          },
        }, () => 'A button!')
      },
    })

    const wrapper = mount(App)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('compound variant', () => {
    const Button = styled('button', {
      backgroundColor: 'gray',
      variants: {
        variant: {
          red: {
            backgroundColor: 'red',
          },
        },
        size: {
          1: {
            height: 30,
          },
          2: {
            height: 60,
          },
        },
      },
      compoundVariants: [
        {
          variant: 'red',
          size: 1,
        },
        {
          backgroundColor: 'compoundColor',
        },
      ],
    })

    const App = defineComponent({
      setup() {
        return () => h('div', [
          h(Button, {
            variant: 'red',
            size: 2,
          }, () => 'not compound'),
          h(Button, {
            variant: 'red',
            size: 1,
          }, () => 'compound'),
        ])
      },
    })

    const wrapper = mount(App)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('breakpoints in variants', () => {
    const { styled: _styled } = createStitches({
      media: {
        bp1: '(min-width: 400px)',
        bp2: '(min-width: 800px)',
      },
    })

    const Button = _styled('button', {
      'color': 'red',
      '@bp1': {
        height: '10px',
      },
      '@bp2': {
        height: '20px',
      },
      'variants': {
        size: {
          big: {
            '@bp1': {
              height: '100px',
            },
          },
        },
      },
    })

    const App = defineComponent({
      setup() {
        return () => h(Button, () => 'no variant')
      },
    })

    const wrapper = mount(App)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('breakpoints in variants and when variant is passed to the component', () => {
    const { styled: _styled } = createStitches({
      media: {
        bp1: '(min-width: 400px)',
        bp2: '(min-width: 800px)',
      },
    })

    const Button = _styled('button', {
      'color': 'red',
      '@bp1': {
        height: '10px',
      },
      '@bp2': {
        height: '20px',
      },
      'variants': {
        size: {
          small: {
            '@bp1': {
              height: '100px',
            },
            '@bp2': {
              height: '200px',
            },
          },
          big: {
            '@bp1': {
              height: '1000px',
            },
            '@bp2': {
              height: '2000px',
            },
          },
        },
      },
    })

    const App = defineComponent({
      setup() {
        return () => h(Button, {
          size: 'big',
        }, () => 'with variant')
      },
    })

    const wrapper = mount(App)

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('breakpoints in variants and when a responsive variant is passed to the component', () => {
    const { styled: _styled } = createStitches({
      media: {
        bp1: '(min-width: 400px)',
        bp2: '(min-width: 800px)',
      },
    })

    const Button = _styled('button', {
      'color': 'red',
      '@bp1': {
        height: '10px',
      },
      '@bp2': {
        height: '20px',
      },
      'variants': {
        size: {
          small: {
            '@bp1': {
              height: '100px',
            },
            '@bp2': {
              height: '200px',
            },
          },
          big: {
            '@bp1': {
              height: '1000px',
            },
            '@bp2': {
              height: '2000px',
            },
          },
        },
      },
    })

    const App = defineComponent({
      setup() {
        return () => h(Button, {
          size: {
            '@bp1': 'small',
          },
        }, () => 'with responsive variant')
      },
    })

    const wrapper = mount(App)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
