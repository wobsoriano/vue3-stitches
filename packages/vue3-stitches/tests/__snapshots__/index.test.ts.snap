// Vitest Snapshot v1

exports[`createStitches Vue > basic component 1`] = `"<h1 class=\\"c-gmqXFB\\"></h1>"`;

exports[`createStitches Vue > breakpoints in variants 1`] = `"<button class=\\"c-fxmqx\\">no variant</button>"`;

exports[`createStitches Vue > breakpoints in variants and when a responsive variant is passed to the component 1`] = `"<button class=\\"c-fxmqx c-fxmqx-ghTNOb-size-small\\" size=\\"[object Object]\\">with responsive variant</button>"`;

exports[`createStitches Vue > breakpoints in variants and when variant is passed to the component 1`] = `"<button class=\\"c-fxmqx c-fxmqx-keNwTR-size-big\\" size=\\"big\\">with variant</button>"`;

exports[`createStitches Vue > component with variant 1`] = `"<div><button class=\\"c-dmZJpl\\">A button!</button><button class=\\"c-dmZJpl c-dmZJpl-dkEfkH-variant-red\\" variant=\\"red\\">A button!</button><button class=\\"c-dmZJpl c-dmZJpl-fNnBJi-variant-blue\\" variant=\\"blue\\">A button!</button></div>"`;

exports[`createStitches Vue > compound variant 1`] = `"<div><button class=\\"c-dmZJpl c-dmZJpl-dkEfkH-variant-red c-dmZJpl-dQBkOa-size-2\\" variant=\\"red\\" size=\\"2\\">not compound</button><button class=\\"c-dmZJpl c-dmZJpl-dkEfkH-variant-red c-dmZJpl-fbRMjr-size-1\\" variant=\\"red\\" size=\\"1\\">compound</button></div>"`;

exports[`createStitches Vue > css prop 1`] = `"<button class=\\"c-dmZJpl c-dmZJpl-igmqXFB-css\\" css=\\"[object Object]\\">A button!</button>"`;
